### PR TITLE
DOP-2303  Add Serverless for Automating Infrastructure management

### PR DESCRIPTION
**Pull request by GuruPKK**
DOP-2303  Add Serverless for Automating Infrastructure management
DOP-2313   Create buckets with right policies

### CODE REVIEW

File: worker/jobTypes/S3Publish.js [hunk 1]
```diff
@@ -98,14 +98,19 @@ class S3PublishClass {
...
+          //contains URLs corresponding to files updated via our push to S3
+          const updatedURLsArray = stdoutJSON.urls;
+          // purgeCache purges the now stale content and requests the URLs to warm the cache for our users
+          logger.save(`${JSON.stringify(updatedURLsArray)}`);
+          await this.fastly.purgeCache(updatedURLsArray);
+          //save purged URLs to job object
+          await workerUtils.updateJobWithPurgedURLs(this.GitHubJob.currentJob, updatedURLsArray);
+        } catch (error) {
+          console.trace(error)
+        }
```
Comment (casthewiz) [inline worker/jobTypes/S3Publish.js:113]: I don't think we want this change merged

**Review by casthewiz — COMMENTED** [review]
I don't think we should delete the old workflows until we verify the new workflows are fully functional and our prod ecs is fully up post merge - afaik there's no conflict between the two?

File: worker/jobTypes/S3Publish.js (hunk 1 above)
Comment (GuruPKK) [reply worker/jobTypes/S3Publish.js:113]: sometimes when purge fails it fails the build as well, so we wanted to treat purge failures as non terminal, thoughts?

Comment (GuruPKK) [reply worker/jobTypes/S3Publish.js]: Yes

**Comment by GuruPKK** [comment]
> I don't think we should delete the old workflows until we verify the new workflows are fully functional and our integration/prod ecs is fully up post merge and release - afaik there's no conflict between the two?

Done

**Review by casthewiz — APPROVED** [review]
We should probably add a section in the Readme.md explaining what's going on with `self.custom` and the `serverless.yml` for anyone who hasn't exposure to serverless 🤔

**Pull request merged**